cpp/ql/src/Security/CWE/CWE-732/FilePermissions.qll

```diff
@@ -15,15 +15,21 @@ int parseHex(string input) {
 }
 
 int o_creat() {
-  exists(Macro m | m.getName() = "O_CREAT" |
-    result = parseHex(m.getBody()) or result = UnixConstants::parseOctal(m.getBody())
-  )
+  result =
+    unique(int v |
```

Comment:
Is this `unique` actually required? Is it ever the case that the body can both be valid hex and valid octal?

Reply:
There's on codebase that that defines `O_CREAT` with two different values. So, not so much hex vs. oct. Maybe worth a comment?

Reply:
Ah, I see! It would be great to have a comment on that, indeed :)

Reply:
I tried to address this. Not sure if I have been clear enough, so please have a look.

Reply:
Awsome. That's exactly how I would phrase it as well :)